File: completed/2419.java
class Solution {
    public int longestSubarray(int[] nums) {
        int maximum = 0;
        for (int num : nums) {
            maximum = Math.max(num, maximum);
        }

        int longest = 1;
        int left = 0;
        int right = 0;
        while (left < nums.length) {
            if (nums[left] == maximum) {
                right = left + 1;
                while (right < nums.length && nums[right] == maximum) {
                    right++;
                }
                longest = Math.max(right - left, longest);
                left = right;
            } else {
                left++;
            }
        }
        return longest;
    }
}

File: completed/2419.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int target = 0;
        for (int num : nums) {
            target = max(num, target);
        }

        int left = 0;
        int right = 0;
        int length = 1;
        while (right < nums.size()) {
            if (nums[right] != target) {
                length = max(right - left, length);
                left = right + 1;
            }
            right++;
        }

        if (left < nums.size()) {
            length = max(right - left, length);
        }

        return length;
    }
};
